### DISCUSSION TITLE
The future of django-background-tasks

### DISCUSSION THREAD

**Discussion by iamjonmiller** [django-background-tasks/django-background-tasks#298] [Announcements]
Hey everyone! I'm a long time user of this wonderful package and I have been maintaining a [fork](https://github.com/iamjonmiller/django-background-tasks) and [working package](https://pypi.org/project/django-background-tasks-updated/) for a few years since the Django 4 update broke the published code. @philippeowagner has generously handed over the keys to this repository so I can continue to maintain compatibility with future Django and Python updates. 

My goals in the near future are to:
- [x] Merge in updated code from my working fork
- [x] Publish update to forked package explaining the change
- [x] Archive forked repo
- [ ] Cleanup this repository by dealing with issues, pull requests, and more
- [x] Publish a new build to the official package

I plan on working on this project in my free time, as I am full time Django dev, but rest assured we use this package in my day job and I will make sure breaking changes get fixed. If you have good ideas for improvements, please feel free to continue to share them and perhaps we can start to integrate upgrades as well. I am following [DEP-14](https://www.djangoproject.com/weblog/2024/may/29/django-enhancement-proposal-14-background-workers/) closely, which is proposing a native form of background workers in Django, and that might play a big part in where this package ends up going in the future.

**Comment by philippeowagner** [comment]
Thank you, @iamjonmiller, for your effort and contributions towards django-background-tasks.

**Comment by ianastewart** [comment]
Hi Jonathan,
Great to hear that your work on this valuable package is going to be merged into the original and that ownership has been resolved. I am also glad to hear that you are following DEP-14. I was considering migrating to their unfinished reference implementation but will hold off for the present in the expectation of good things in the future.
Regards
Ian Stewart

**Comment by saqibur** [comment]
Hello @iamjonmiller hope you're doing well. Is this package still being maintained?

**Reply by iamjonmiller** [reply]
> Hello @iamjonmiller hope you're doing well. Is this package still being maintained?

Yes, but I am currently evaluating what can be replaced by the native background tasks in django 6. I still run this every day on my production servers.

**Reply by saqibur** [reply]
That's good to hear! @iamjonmiller . I'm using it in a few projects as well in production and absolutely love the package.

About background tasks in Django 6, I think it's going to be a while before things pick up and we'll have the complete API + stability + more pluggable backends to get things going. Maybe Django 6.2? 

I wanted to ask two things:
1. Are there any plans to adapt this package to use the "new" tasks framework in Django, since this can essentially be the DB backend for it?
2. Since I've been using this package a lot and quite often recommend it to others, I was wondering if there is any way I could contribute to it? One of the things that I believe could be improved are some more examples and documentation on how to use the package.

I really appreciate your reply, I hope I'm not wasting your time with these questions. Thanks again!

**Reply by iamjonmiller** [reply]
You aren't wasting my time. My apologies for the slow responses as I am quite busy at work and with a new kid.

1. Presently no plans, but I am very open to proposals.
2. Please feel free to contribute with pull requests or detailed ideas! The documentation has always been a weak spot for this package and I remember all those years ago when I was trying to pick up the piece and get this thing working again.

**Reply by saqibur** [reply]
Following up on this, and starting small @iamjonmiller - https://github.com/django-background-tasks/django-background-tasks/issues/312

Please share your thoughts and let me know if I should process. Thanks!
